src/tcgwars/logic/impl/gen3/PopSeries2.groovy

```diff
@@ -245,34 +279,61 @@ public enum PopSeries2 implements LogicCardInfo {
       return supporter (this) {
         text "Choose 1 of your Pokémon in play (excluding Pokémon ex). Return that Pokémon and all cards attached to it to your hand."
         onPlay {
+          def validTargets = my.all.findAll { !it.EX }
+
+          def tar = validTargets.select("Which Pokémon to put back into your hand?")
+          targeted(tar, Source.TRAINER_CARD) {
+            tar.cards.moveTo(my.hand)
+            removePCS(tar)
+          }
         }
-        playRequirement{
+        playRequirement {
+          assert my.all.findAll { !it.EX } : "No eligible Pokémon to return back to your hand."
         }
       };
       case MULTI_TECHNICAL_MACHINE_01_9:
       return basicTrainer (this) {
-        text "The Defending Pokémon is now Paralyzed."
-        onPlay {
+        text "Attach this card to 1 of your Pokemon in play. This Pokemon may use this card's attack instead of its own. At the end of your turn, discard Multi Technical Machine 01"
+        def eff
+        onPlay {reason->
+          def moveBody = {
+            text "The Defending Pokémon is now Paralyzed."
+            energyCost C
+            attackRequirement {}
+            onAttack {
+              apply PARALYZED
+            }
+          }
+          Move move=new Move("Paralyzing Gaze")
+          moveBody.delegate=new MoveBuilder(thisMove:move)
+          moveBody.call()
+          eff = getter GET_MOVE_LIST, self, {h->
+            h.object.add(move)
+          }
         }
-        playRequirement{
+        onRemoveFromPlay {
+          eff.unregister()
         }
       };
       case POKEMON_PARK_10:
       return stadium (this) {
         text "Once during each player's turn, when that player attaches an Energy card from his or her hand to 1 of his or her Benched Pokémon, he or she removes 1 damage counter from that Pokémon."
+        def eff
         onPlay {
+          eff = delayed {
+            after ATTACH_ENERGY, {
+              if (ef.reason == PLAY_FROM_HAND) {
```

Comment:
This seems okay, but to throw a wrench in here ...

When the card says "when that player attaches an Energy card from his or her hand" does it mean so long as the card was in the hand before being attached, like Rare Candy does with evolution, and evolution preventing abilities check for to prevent Rare Candy?

If so you probably need a flag that resets at the end of the turn, and check explicitly if the card was in hand before being attached instead of relying on the `reason`.

Reply:
Another way to so it would be something like 
```suggestion
              def card
              before ATTACH_ENERGY, {
              if(my.hand.contains(ef.card){
              card = ef.card
              }
              }
              after ATTACH_ENERGY, {
              if(ef.card = card){
              heal 10, ef.resolvedTarget
              }
              }
```